src/library/CMakeLists.txt

```
@@ -3,35 +3,33 @@ cmake_minimum_required(VERSION 3.13)
 
 add_library(${PROJECT_NAME} STATIC
   geodesy/geodetic_position.cpp
-  
-  igrf/igrf.cpp
 
-  inih/ini.c
-  inih/cpp/INIReader.cpp
+  randomization/global_randomization.cpp
+  randomization/normal_randomization.cpp
```

```suggestion
  randomization/global.cpp
  randomization/normal.cpp
```
randomization というディレクトリを切るならこれぐらいシンプルでもいいのではないでしょうか

---

現状S2Eは、ファイル名とクラス名が一致する形式を取っているので、ファイル名を一般性の高いものにするのは違う気がしていますがいかがでしょう？

---

完全にファイル名とクラス名を一致させなければならないとするとそれこそ上の `_functions.hpp` とかは（まあ前方一致はしてますが）どうなんだろうという気もしますし，そこまで拘らなくてもいいのではとは思います．
あとは，せっかくディレクトリを切っているので名前空間を切って，その上でクラス名そのものは `Global` とかにしてしまうとかもあり得そうです．
この文脈での global と normal の意図があんまりわかっていないのでちょっと頓珍漢なこと言ってるかもですが．

---

`vector_template_functions.hpp`は、`vector.hpp`で定義されている`Vector`クラスの具体的な実装をしているだけなので、ファイル名完全一致の話とは別だと思っています。  

今のS2Eの命名の雰囲気の中で、このクラス名そのものをGlobalにするのはかなり違和感あるので、やるとなっても今すぐではなくもう少し議論を深めてからにしたいですね。ここだけ変えても違和感が残るので、全てのクラス名の命名方針を変えるとかになる気がします。

---

例えば、`GlobalEnvironment`の命名方法にも影響したりします。

---

なるほど．個人的には
- ディレクトリを切る程度にまとめて扱いたいものは名前空間を切った方がよい（名前の prefix で頑張るのではなく）
- 過度に`using`しなければそんなに違和感はないのでは（ex: `randomization::Global` みたいなかんじで使う）

と思ったんですがどうでしょう

---

と書いてから global とかは名前の prefix ではなく suffix なことに気付いたので微妙かも

---

どこかの段階でに
```
randomization/global.hpp
environment/global.hpp
```
というファイル構成で
```
randomization::Global
environment::Global
```
というクラスで進めていくということでしょうか。ディレクトリ分けされているものの`global.hpp`が2つあるのは少し違和感があるきがしています。

---

です．単なるクラス（と合わせるファイルも）の名前は名前空間から適切に文脈を判断できれば問題ないと思っています（ex: `std::vector` , `math::vector`）．